ksql-engine/src/test/java/io/confluent/ksql/function/udf/json/JsonArrayContainsTest.java

```diff
@@ -0,0 +1,103 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.json;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+public class JsonArrayContainsTest
+{
+    private JsonArrayContains jsonUdf = new JsonArrayContains();
+
+    @Test
+    public void shouldReturnFalseOnEmptyArray() {
+        assertEquals(false, jsonUdf.contains("[]", true));
+        assertEquals(false, jsonUdf.contains("[]", false));
+        assertEquals(false, jsonUdf.contains("[]", null));
+        assertEquals(false, jsonUdf.contains("[]", 1.0));
+        assertEquals(false, jsonUdf.contains("[]", 100));
+        assertEquals(false, jsonUdf.contains("[]", "abc"));
+        assertEquals(false, jsonUdf.contains("[]", ""));
+    }
+
+    @Test
+    public void shouldNotFindValuesInNullArray() {
+        assertEquals(true, jsonUdf.contains("[null]", null));
+        assertEquals(false, jsonUdf.contains("[null]", "null"));
+        assertEquals(false, jsonUdf.contains("[null]", true));
+        assertEquals(false, jsonUdf.contains("[null]", false));
+        assertEquals(false, jsonUdf.contains("[null]", 1.0));
+        assertEquals(false, jsonUdf.contains("[null]", 100));
+        assertEquals(false, jsonUdf.contains("[null]", "abc"));
+        assertEquals(false, jsonUdf.contains("[null]", ""));
+    }
+
+    @Test
+    public void shouldFindIntegersInJsonArray() {
+        final String json = "[2147483647, {\"ab\":null }, -2147483648, 1, 2, 3, null, [4], 4]";
+        assertEquals(true, jsonUdf.contains(json, 2147483647));
+        assertEquals(true, jsonUdf.contains(json, -2147483648));
+        assertEquals(true, jsonUdf.contains(json, 1));
+        assertEquals(true, jsonUdf.contains(json, 2));
+        assertEquals(true, jsonUdf.contains(json, 3));
+        assertEquals(false, jsonUdf.contains("5", 5));
+        assertEquals(false, jsonUdf.contains(json, 5));
+    }
+
+    @Test
+    public void shouldFindLongsInJsonArray() {
+        assertEquals(true, jsonUdf.contains("[1]", 1L));
+        assertEquals(true, jsonUdf.contains("[1111111111111111]", 1111111111111111L));
+        assertEquals(true, jsonUdf.contains("[[222222222222222], 33333]", 33333L));
+        assertEquals(true, jsonUdf.contains("[{}, \"abc\", null, 1]", 1L));
+        assertEquals(false, jsonUdf.contains("[[222222222222222], 33333]", 222222222222222L));
+        assertEquals(false, jsonUdf.contains("[{}, \"abc\", null, [1]]", 1L));
+        assertEquals(false, jsonUdf.contains("[{}, \"abc\", null, {\"1\":1}]", 1L));
+    }
+
+    @Test
+    public void shouldFindDoublesInJsonArray() {
+        assertEquals(true, jsonUdf.contains("[-1.0, 2.0, 3.0]", 2.0));
```

Comment:
What about
```
jsonUdf.contains("[-1.0, 2, 3.0]", 2.0)
```
and
```
jsonUdf.contains("[-1.0, 2.0, 3.0]", 2)
```
? 

I assume both return false?

Reply:
There's a test for this at the end of `shouldFindDoublesInJsonArray` but I added an extra one for longs.